designer/client/context/__tests__/FeatureFlagContext.jest.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { Router } from "react-router-dom";
+import { act } from "react-dom/test-utils";
+import { FeatureFlagProvider } from "../FeatureFlagContext";
+import { createMemoryHistory } from "history";
+import sinon from "sinon";
+import { FeatureToggleApi } from "../../api/toggleApi";
+import FeatureToggle from "../../FeatureToggle";
+import { useFeatures } from "../../hooks/featureToggling";
+const history = createMemoryHistory();
+history.push("");
+
+const customRender = (ui, { providerProps, ...renderOptions }) => {
+  const rendered = render(
+    <Router history={history}>
+      <FeatureFlagProvider value={providerProps}>{ui}</FeatureFlagProvider>
+    </Router>,
+    renderOptions
+  );
+  return {
+    ...rendered,
+    rerender: (ui, options) =>
+      customRender(ui, { container: rendered.container, ...options }),
+  };
+};
+
+describe("FeatureFlagContext", () => {
+  beforeEach(() => {
+    sinon.restore();
```

Comment:
no need to acton, but jest should cleanup mocks/spies if it makes your life easier

Reply:
<img width="1458" alt="Screenshot 2021-02-02 at 11 41 01" src="https://user-images.githubusercontent.com/3599604/106595484-8ac43880-654b-11eb-8d80-c042b38e84d4.png">

I did want to remove but looks like its not great at the cleanup

Reply:
sorry, I meant that jest mocks `jest.fn()` vs sinon - iirc jest should restore jest mocks after each test

Reply:
Good point, have we decided to move away from sinon on our move towards jest?

Reply:
oops, sorry for late reply, yep